internal/app/singularity/cache_list_linux.go

```diff
@@ -0,0 +1,222 @@
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			printFileSize, err := findSize(fileInfo.Size())
+			if err != nil {
+				// no need to describe the error, since it is already
+				sylog.Warningf("%v", err)
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "oci")
+			printFileSize = ""
+		}
+	}
+	return nil
+}
+
+func listBlobCache(printList bool) error {
+	// loop thrught ociBlob cache
+	count := 0
+	var totalSize int64
+
+	_, err = os.Stat(filepath.Join(cache.OciBlob(), "/blobs"))
+	if os.IsNotExist(err) {
+		return nil
+	}
+	blobs, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/"))
+	if err != nil {
+		sylog.Warningf("Unable to opening oci folder: %v", err)
+		return err
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/", f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciBlob(), "/blobs/", f.Name(), b.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			if printList == true {
+				printFileSize, err := findSize(fileInfo.Size())
+				if err != nil {
+					// no need to describe the error, since it is already
+					sylog.Warningf("%v", err)
+				}
+				fmt.Printf("%-22.20s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "blob")
+				printFileSize = ""
+			}
+			count++
+			totalSize += fileInfo.Size()
+		}
+	}
+	if printList != true && count >= 1 {
+		printFileSize, err := findSize(totalSize)
+		if err != nil {
+			// no need to describe the error, since it is already
+			sylog.Warningf("%v", err)
+		}
+		fmt.Printf("\nThere are %d oci blob file(s) using %v of space. Use: '-T=blob' to list\n", count, printFileSize)
+		printFileSize = ""
+	}
+	return nil
+}
+
+// ListSingularityCache : list local singularity cache, typeNameList : is a string of what cache
+// to list (seprate each type with a comma; like this: library,oci,blob) allList : force list all cache.
+func ListSingularityCache(typeNameList string, allList bool) error {
+	libraryList := false
+	ociList := false
+	blobList := false
+	listBlobSum := false
+
+	// split the string for each `,` then loop throught it and find what flags are there.
+	// then see whats true/false later. heres the benefit of doing it like this; if the user
+	// specified `library` twice, it will still only be printed once.
+	if len(typeNameList) >= 1 {
+		for _, nameType := range strings.Split(typeNameList, ",") {
+			switch nameType {
+			case "library":
+				libraryList = true
+			case "oci":
+				ociList = true
+			case "blob", "blobs":
+				blobList = true
+			case "all":
+				allList = true
+			default:
+				sylog.Fatalf("Not a valid type: %v", nameType)
+				os.Exit(2)
+			}
+		}
+	} else {
```

Comment:
I'm confused at the purpose here. Are we saying that if `typeNameList == ""` it should list all?

Reply:
Yes, if `typeNameList` is empty (`""`), then list `library`, and `oci` cache. Blobs are not listed by default (unless type blob is specified)

Reply:
OK, so in this case I think there's a better approach for handling defaults. I will leave a few additional review comments for how we can more cleanly do this

Reply:
We can now just remove this `} else {...}`